### PR TITLE
[5.5.x] Bump go to 1.17.5

### PR DESCRIPTION
**Pull request by bernardjkim**
## Description
Bump go version to `1.17.5`.
Applied some changes that were required due to [module changes in Go 1.16](https://go.dev/blog/go116-module-changes).
- `go install` now requires a version suffix.
- Go defaults to `GO111MODULE=on`. We need to set `GO111MODULE=off` in places we want to continue using GOPATH mode.

## Linked tickets and PRs
- Ports https://github.com/gravitational/planet/pull/871
- Refs https://github.com/gravitational/gravity/issues/2693

### CODE REVIEW

File: build.assets/docker/base.dockerfile [hunk 1]
```diff
@@ -89,7 +89,7 @@ RUN curl https://download.docker.com/linux/static/stable/x86_64/docker-$DOCKER_V
     cp /tmp/docker/* /usr/bin && \
     rm -rf /tmp/docker*
 
-RUN curl https://storage.googleapis.com/kubernetes-helm/helm-$HELM_VER-linux-amd64.tar.gz -o /tmp/helm-$HELM_VER.tar.gz && \
+RUN curl https://get.helm.sh/helm-$HELM_VER-linux-amd64.tar.gz -o /tmp/helm-$HELM_VER.tar.gz && \
```
Comment (wadells) [inline build.assets/docker/base.dockerfile:92]: This doesn't look related... it isn't in any of the earlier ports.  Whats up?

Comment (bernardjkim) [reply build.assets/docker/base.dockerfile:92]: https://helm.sh/blog/get-helm-sh/
This repo was already updated in later versions of Gravity. Attempting to download helm using googleapis.com now fails with
```
<?xml version='1.0' encoding='UTF-8'?><Error><Code>AccessDenied</Code><Message>Access denied.</Message><Details>Anonymous caller does not have storage.objects.get access to the Google Cloud Storage object.</Details></Error>%
```
I could split into a separate PR to keep PRs consistent.

Comment (bernardjkim) [reply build.assets/docker/base.dockerfile:92]: Oh gotcha. Looks like the download url was updated in https://github.com/gravitational/planet/pull/787. But the PR doesn't really provide much context for the changes.

**Review by wadells — APPROVED** [review]

**Review by knisbet — APPROVED** [review]
